frontend/cypress/integration/common/graph_manipulation.ts

```diff
@@ -0,0 +1,69 @@
+import { Then, And, When } from "cypress-cucumber-preprocessor/steps";
+
+function buttonClick(label:string){
+    cy.get(`button[aria-label="${label}"]`).click();
+};
+
+function buttonPrepare(label:string, active:boolean){
+    cy.waitForReact();
+    cy.getReact("Button",{ props: { 'aria-label': `${label}`}})
+    .should('have.length', '1')
+    .getProps()
+    .then((props => {
+        if (props.isActive !== active){
+            buttonClick(label);
+        }
+    }));
+};
+
+function buttonState(label:string, active:boolean){
+    cy.waitForReact();
+    cy.getReact("Button",{ props: { 'aria-label': `${label}`}})
+    .should('have.length', '1')
+    .getProps('isActive')
+    .should('eq', active);
+}
+
+Then("the toggle button {string} is enabled", (label: string) => {
+    cy.get(`button[aria-label="${label}"]`).should('have.attr', 'aria-disabled', 'false');
+});
+
+And("the toggle button {string} is enabled", (label: string) => {
+    cy.get(`button[aria-label="${label}"]`).should('have.attr', 'aria-disabled', 'false');
+});
+
+And("the button {string} is clicked", (label: string) => {
+    buttonClick(label);
+});
+
+Then("the button {string} is active", (label: string) => {
+   buttonState(label, true); 
+});
+
+Then("the button {string} is not active", (label: string) => {
+   buttonState(label, false); 
+});
+
+And("the {string} is turned on", (label: string) => {
+   buttonPrepare(label, true); 
+});
+
+And("the {string} is turned off", (label: string) => {
+   buttonPrepare(label, false); 
+});
+
+Then("user can see the legend section", () => {
+    cy.get("span").contains("Legend").should("be.visible");
```

Comment:
It'd be better to use something more specific than looking at all the span elements for the text "Legend". If there's multiple legends on the screen this could fail later or be flaky. This is a good case for adding a `data-test` attribute to the legend and searching for that. There's a `GraphLegend` component you can add it to: https://github.com/kiali/kiali/blob/master/frontend/src/pages/Graph/GraphLegend.tsx#L44.

Reply:
I tried it and I am not sure if I did it the right way, because it broke one of the CI pipelines.

Reply:
It looks like just a snapshot test that needs to be updated. When you run the ui unit tests locally you should see an option for updating those.